File: lawpaypro/shared/src/main/java/rs/wepublishlaws/shared/messages/PaymentResponse.java
package rs.wepublishlaws.shared.messages;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.RequiredArgsConstructor;
import rs.wepublishlaws.shared.PaymentStatus;
import rs.wepublishlaws.shared.SdkParamsDto;

@Data
@RequiredArgsConstructor
@AllArgsConstructor
public class PaymentResponse {
    private String paymentUrl;
    private String paymentId;
    private PaymentStatus status;
    private SdkParamsDto sdkParams;
    private String errorMessage;
    private String errorCode;
}

File: law-pay-pro-fe/src/app/pages/payments/responses/payment-response.ts
export interface PaymentResponseDto {
  paymentUrl: string;
  paymentId: string;
  sdkParams: SdkParamsDto;
  errorMessage: string;
  errorCode: string;
}

export interface SdkParamsDto {
  qrCode: string;
  paymentAddress: string;
  tag: string;
}
